ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/UserManagementServiceIT.scala

```diff
@@ -263,99 +259,133 @@ final class UserManagementServiceIT extends LedgerTestSuite {
-    val user1 = User(userId1, "")
-    val user2 = User(userId2, "")
-    val user3 = User(userId3, "")
-    val user4 = User(userId4, "")
-    val user5 = User(userId5, "")
-    val user6 = User(userId6, "")
+
     for {
-      // Ensure we have at least 6 users:
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user1), Nil))
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user2), Nil))
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user3), Nil))
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user4), Nil))
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user5), Nil))
-      _ <- ledger.userManagement.createUser(CreateUserRequest(Some(user6), Nil))
-      // Requesting first page:
-      res1 <- ledger.userManagement.listUsers(ListUsersRequest(pageSize = 2, pageToken = ""))
-      // Requesting second page:
-      res2 <- ledger.userManagement.listUsers(
-        ListUsersRequest(pageSize = 3, pageToken = res1.nextPageToken)
-      )
-      // Requesting last non-empty page of users
-      res3 <- ledger.userManagement.listUsers(
-        ListUsersRequest(pageSize = 1000, pageToken = res2.nextPageToken)
-      )
-      // Using not base64 encoded string as a page token
-      onBadTokenError <- ledger.userManagement
-        .listUsers(
-          ListUsersRequest(pageSize = 100, pageToken = UUID.randomUUID().toString)
+      // Create 4 users to ensure we have at least two pages of two users each
+      _ <- ledger.createUser(CreateUserRequest(Some(User(userId1, "")), Nil))
+      _ <- ledger.createUser(CreateUserRequest(Some(User(userId2, "")), Nil))
+      _ <- ledger.createUser(CreateUserRequest(Some(User(userId3, "")), Nil))
+      _ <- ledger.createUser(CreateUserRequest(Some(User(userId4, "")), Nil))
+      // Fetch the first two full pages
+      page1 <- ledger.userManagement.listUsers(ListUsersRequest(pageToken = "", pageSize = 2))
+      page2 <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageToken = page1.nextPageToken, pageSize = 2)
+      )
+      // Verify that the second page stays the same even after we have created a new user that is lexicographically smaller then the last user on the first page
+      // (Note: "!" is the smallest valid user-id character)
+      newUserId = "!" + page1.users.last.id
+      _ <- ledger.createUser(CreateUserRequest(Some(User(newUserId)), Seq.empty))
+      page2B <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageToken = page1.nextPageToken, pageSize = 2)
+      )
+      _ = assertEquals("after creating new user before the second page", page2, page2B)
+      // cleanup: deleting new user
+      _ <- ledger.userManagement.deleteUser(DeleteUserRequest(newUserId))
+      // Verify that the second page stays the same even after we have deleted the last user from the first page page
+      _ <- ledger.userManagement.deleteUser(DeleteUserRequest(page1.users.last.id))
```

Comment:
We don't know what user was the last on the first page. It could have been a resident user not created by this test. We cannot delete it

Reply:
Ah, yes! Deleting this section of the test case as I think it's not crucial to test it given all the constraints of conformance tests and we have a more fundamental check "TestListUsersVisibilityOfNewUserWhenCreatedAndThenDeleted" in place. 

Also configuring this test case run serially.